linked_list/implementation.py

```diff
@@ -1,40 +1,90 @@
 from .interface import AbstractLinkedList
+from .node import Node
 
+# Don't think I'm as far off as the tests make it seem 
+# Believe the append is working correctly once the TypeError and eq are cleaned up
+<<<<<<< HEAD
+# Haven't started pop functionality because I know I'll need get_item
+# Can't test get_item until 'NoneType' issue is resolved
+=======
+# Haven't started pop functionality
+#
+>>>>>>> 2eb5873dc8c0c4940ff240a611eb5ec49b5c35ae
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
-
+        # if elements defaulted to None start and end (needed in tests) default to None as well for error handling
+        self.start = None
+        self.end = None
+        # Now where/how to populate these? >> assuming the append and iter come into play here >> yep.
+        # Need to build list of nodes using append starting at start until >> self.start is None b/c out of elements
+        # elements constructor
+        self.elements = [self.append(elem) for elem in elements if elements]
+        
     def __str__(self):
-        pass
+        #generate string off a list
+        _str = ', '.join([str(elements) for elements in self])
+        #match formatting
+        return '[{}]'.format(_str)
 
     def __len__(self):
-        pass
+        return self.count()
 
     def __iter__(self):
-        pass
+        #need to make self start None to Stop Iteration when self.start.next is None
+        while self.start:
+            yield self.start.elem
+            self.start = self.start.next
+        raise StopIteration
 
     def __getitem__(self, index):
+        '''
+        return self.elements[index]
+        '''
         pass
 
     def __add__(self, other):
+        '''
+        if isinstance(other, self.__class__):
+            result_list = self.__class__([elem for elem in self.elements])
+            result_list.append([elem for elem in other.elements])
+        return result_list
+        '''
         pass
 
     def __iadd__(self, other):
+        '''
+        return self + other
+        '''
         pass
 
     def __eq__(self, other):
-        pass
+        if isinstance(other, self.__class__):
+            return self.elements == other.elements
```

Comment:
This is ringing some bells in my mind... You're creating `self.elements` in the constructor. What happens if you modify a list? Can you add that test and see?

```python
l1 = LinkedList([1, 2, 3])
l2 = LinkedList([1, 2, 3])

l2.append(4)

assert not l1 == l2
```